tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+# Copyright (c) 2024 Alibaba Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
+
+    setup_common
+    UNSIGNED_UNPROTECTED_REGISTRY_IMAGE="quay.io/prometheus/busybox:latest"
+    UNSIGNED_PROTECTED_REGISTRY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:unsigned"
+    COSIGN_SIGNED_PROTECTED_REGISTRY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:cosign-signed"
+    COSIGNED_SIGNED_PROTECTED_REGISTRY_WRONG_KEY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:cosign-signed-key2"
+}
+
+function setup_kbs_image_policy() {
+    if [ "${KBS}" = "false" ]; then
+        skip "Test skipped as KBS not setup"
+    fi
+
+    default_policy="${1:-insecureAcceptAnything}"
+    policy_json=$(cat << EOF
+{
+    "default": [
+        {
+        "type": "${default_policy}"
+        }
+    ],
+    "transports": {
+        "docker": {
+            "ghcr.io/confidential-containers/test-container-image-rs": [
+                {
+                    "type": "sigstoreSigned",
+                    "keyPath": "kbs:///default/cosign-public-key/test"
+                }
+            ],
+            "quay.io/prometheus": [
+                {
+                    "type": "insecureAcceptAnything"
+                }
+            ]
+        }
+    }
+}
+EOF
+    )
+
+    public_key=$(cat<<EOF
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwQEjdCiL3ILUf07NDkDVhgKCj1C6
+BsCfmM/zt1kNSj0/+nAqA+25XfyClYq2lJFJ6TkgCsf57cTCkXYDz9c+Yg==
+-----END PUBLIC KEY-----
+EOF
+)
+
+    if ! is_confidential_hardware; then
+        kbs_set_allow_all_resources
+    fi
+
+    kbs_set_resource "default" "security-policy" "test" "${policy_json}"
+    kbs_set_resource "default" "cosign-public-key" "test" "${public_key}"
+}
+
+@test "Create a pod from an unsigned image, on an insecureAcceptAnything registry works" {
+    # We want to set the default policy to be reject to rule out false positives
+    setup_kbs_image_policy "reject"
+
+    create_coco_pod_yaml "${UNSIGNED_UNPROTECTED_REGISTRY_IMAGE}" "kbs:///default/security-policy/test" "" "" "resource" "$node"
```

Comment:
Would it be better to set a variable like `IMAGE_POLICY_URL` to replace the value "kbs:///default/security-policy/test" in the tests?

Reply:
Done, thanks for the suggestion